index.d.ts

```diff
@@ -1161,7 +1161,19 @@ declare namespace plugins {
      * the key/value pairs to record. For example, using
      * `variables => variables` would record all variables.
      */
-    variables?: string[] | ((variables: { [key: string]: any }) => { [key: string]: any });
+    variables?:
+      | string[]
+      | ((
+          variables: { [key: string]: any },
+          info: {
+            span?: Span;
+            args?: ExecutionArgs;
+            operation?: {
+              name: string;
+              operation: "query" | "mutation" | "subscription";
```

Comment:
What's the difference between `operation.name` and `operation.operation`? Can the latter be renamed to something else?

Reply:
Actually, looking at the code it seems to be receiving `type` and not `operation`?

Reply:
Fixed in 21e54d0a6c659b51b0de52879ea0693d56b2ebed, also updated the types to reflect optionality correctly.